applications/DEM_application/custom_strategies/strategies/explicit_solver_strategy.cpp

```diff
@@ -381,30 +381,6 @@ namespace Kratos {
 
             if (number_of_conditions > 0) {
                 CheckHierarchyWithCurrentNeighbours();
-                const int number_of_conditions = (int) pTConditions.size();
-                const int number_of_particles = (int) mListOfSphericParticles.size();
-
-                #pragma omp parallel
-                {
-                    #pragma omp for
-                    for (int i = 0; i < number_of_conditions; i++) {
-                        ConditionsArrayType::iterator ic = pTConditions.begin() + i;
-                        DEMWall* wall = dynamic_cast<Kratos::DEMWall*> (&(*ic));
-                        wall->mNeighbourSphericParticles.resize(0);
-                    }
-
-                    #pragma omp for
-                    for (int i = 0; i < number_of_particles; i++) {
-                        for (unsigned int j = 0; j < mListOfSphericParticles[i]->mNeighbourRigidFaces.size(); j++) {
-                            DEMWall* p_wall = mListOfSphericParticles[i]->mNeighbourRigidFaces[j];
-                            #pragma omp critical
-                            {
-                                p_wall->mNeighbourSphericParticles.push_back(mListOfSphericParticles[i]);
-                            }
-                        }
-                    }
-                }//#pragma omp parallel
-
```

Comment:
Why do you think this is not necessary? If the wall neighbours change, the walls should also be aware that the neighbour particles change.

Reply:
Yes, it is a mistake that I made thinking about the previous design. Now the neighbours do change potentially at every step.